docs/tasking_system/tasking_system.md

```diff
@@ -0,0 +1,74 @@
+#  Tasking System
+The tasking system enables our app to complete actions, i.e. tasks, asynchronously from the backend API.
+For example, when creating a repository, a “snapshot” task is triggered, and begins to run in parallel to the API server.
+
+For a detailed overview of the tasking system see [this video](https://drive.google.com/file/d/1clTBLDOhPaTmxEd2PUuvmuHYu2oP6Yp1/view) and the [accompanying slide deck](https://docs.google.com/presentation/d/19nzcEvXNS1OzwRmGJ7tPdeiJS2LAbjwMi0YBH-yU-eE/edit#slide=id.p).
+
+
+## Features 
+* Queue and process asynchronous tasks
+* Check the status of queued tasks
+* Requeue a task if its worker times-out or exits early, with a backoff timer
+* Schedule dependent tasks
+* Cancel a task
+* Set a task's priority
+
+## Directory structure
+
+| Package            | Description                                                                                     |
+|--------------------|-------------------------------------------------------------------------------------------------|
+| pkg/tasks          | each file contains code for handling a particular task type                                     |
+| pkg/tasks/queue    | queue used by client and worker to schedule tasks                                               |
+| pkg/tasks/client   | an interface to enqueue a task                                                                  |
+| pkg/tasks/worker   | an interface to dequeue and handle tasks                                                        |
+| pkg/tasks/payloads | workaround to import certain payloads to dao layer, but payloads are not generally defined here |
+
+## Components
+
+### Queue
+
+`Queue` is an interface used by the client and worker packages for scheduling tasks. It is meant to be used through client/worker, not imported independently.
```

Comment:
Re "client/worker"
The use of slash in text is ambiguous and should be avoided. Can you reword that? Do you mean "client or worker" or "client and worker"?

Reply:
thanks, changed it to "or"